operator/api/v1alpha1/keda_types.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/kyma-project/module-manager/operator/pkg/types"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:validation:Enum=debug;info;error
+type LogLevel string
+
+// +kubebuilder:validation:Enum=json;console
+type LogFormat string
+
+// +kubebuilder:validation:Enum=epoch;millis;nano;iso8601;rfc3339;rfc3339nano
+type LogTimeEncoding string
+
+type LoggingOperatorCfg struct {
+	Level        *LogLevel        `json:"level,omitempty"`
+	Format       *LogFormat       `json:"format,omitempty"`
+	TimeEncoding *LogTimeEncoding `json:"timeEncoding,omitempty"`
+}
+
+type LoggingMetricsSrvCfg struct {
+	Level *LogLevel `json:"level,omitempty"`
+}
+
+type LoggingCfg struct {
+	Operator      *LoggingOperatorCfg   `json:"operator,omitempty"`
+	MetricsServer *LoggingMetricsSrvCfg `json:"metricsServer,omitempty"`
+}
+
+type Resources struct {
+	Operator      *corev1.ResourceRequirements `json:"operator,omitempty"`
+	MetricsServer *corev1.ResourceRequirements `json:"metricsServer,omitempty"`
+}
+
+// KedaSpec defines the desired state of Keda
+type KedaSpec struct {
+	Logging   *LoggingCfg        `json:"logging,omitempty"`
+	Resources *Resources         `json:"resources,omitempty"`
+	Env       *map[string]string `json:"env,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// Keda is the Schema for the kedas API
+type Keda struct {
```

Comment:
- I think we should be consistent on naming - [lifecycle-manager](https://github.com/kyma-project/lifecycle-manager/blob/8a95a5ca90dd9f2e63fe8b3e1db66e04e85ee507/operator/api/v1alpha1/kyma_types.go#L243).

Reply:
I don't think both types are the same. The lifecycle manager is not a component operator. It doesn't manage configuration for concrete components, it just deploys other operators. And it it's owned by "us". FWIW I think the type should be renamed there, too.

The the keda manager is a component operator, it's responsible for directly _configuring_  the keda deployment. We should make that purpose as clear as possible. Also, we don't own keda and we don't want to influence their naming. Using a very generic name like this will probably cause confusion and/or naming conflicts/collisions in the future. We should try to be as explicit and specific as possible.

Reply:
- The import path  is `github.com/kyma-project/keda-manager/operator/api/v1alpha1` what is confusing about it and how can you have conflict with it?.
- `KedaManagerConfig` can be also confusing and have collisions in future